bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/hbasesnapshots/ImportHBaseSnapshotJob.java

```diff
@@ -198,6 +204,40 @@ protected static int setConfFromArgs(Configuration conf, String[] args) {
     return 0;
   }
 
+  private static int computeSplitsPerRegion(Configuration configuration) throws IOException {
+    int numSplitsPerRegion = configuration.getInt(SNAPSHOT_SPLITS_PER_REGION_KEY, 0);
+    if (numSplitsPerRegion > 0) {
+      return numSplitsPerRegion;
+    }
+
+    Path hbaseRoot = CommonFSUtils.getRootDir(configuration);
+    String snapshotName = configuration.get(SNAPSHOTNAME_KEY);
+    FileSystem fileSystem = hbaseRoot.getFileSystem(configuration);
+    long targetSplitSize = configuration.getLong(SNAPSHOT_SPLIT_TARGET_SIZE_KEY,
+        SNAPSHOT_SPLIT_TARGET_SIZE_DEFAULT);
+    SnapshotManifest manifest = TableSnapshotInputFormatImpl.getSnapshotManifest(configuration,
+        snapshotName, hbaseRoot, fileSystem);
+
+    long maxRegionSize = 0;
+
+    for (SnapshotRegionManifest regionManifest : manifest.getRegionManifests()) {
+      long regionSize = 0;
+      for (FamilyFiles familyFiles : regionManifest.getFamilyFilesList()) {
+        for (StoreFile storeFile : familyFiles.getStoreFilesList()) {
+          regionSize += storeFile.getFileSize();
+        }
+      }
+      maxRegionSize = Math.max(maxRegionSize, regionSize);
+    }
+
+    numSplitsPerRegion = (int) Math.ceil(maxRegionSize / (float) targetSplitSize);
+    if (numSplitsPerRegion < 1) {
+      numSplitsPerRegion = 1;
+    }
```

Comment:
can we provide an upper bound of splits and set a max if the total splits for the job (regions * splits per region) exceeds this maximum. It would be safe to use 1 million splits for an upper bound. For example: 

```
max splits = 1million 
total splits = regions * splits per region

if (total splits > max splits ) 
 splits per region = max splits  / num of region
```